src/tokenizer/tokenizer.cc

```diff
@@ -15,22 +26,26 @@ TfmStatus CreateBPETokenizer(const std::string& tokenizer_path,
   auto config_file = tokenizer_path + "/tokenizer_config.json";
   auto status = token_cfg->LoadJson(config_file);
 
-  if (tokenizer_type == "BPE") {
-    token_ptr = std::make_unique<BPETokenizer>();
-  } else if (tokenizer_type.empty()) {
-    if (token_cfg->tokenizer_class_ == "LlamaTokenizer" ||
-        token_cfg->tokenizer_class_ == "GemmaTokenizer") {
-      token_ptr = std::make_unique<SpmTokenizer>();
-    } else if (token_cfg->tokenizer_class_ == "GPT2Tokenizer" ||
-               token_cfg->tokenizer_class_ == "CLIPTokenizer" ||
-               token_cfg->tokenizer_class_ == "CodeGenTokenizer") {
-      token_ptr = std::make_unique<BPETokenizer>();
+  std::string type = tokenizer_type;
+  if (type.empty()) {
+    if (BPETokenizer::IsSupportedModel(GetModelName(token_cfg->tokenizer_class_))) {
+      type = "BPE";
+    } else if (std::filesystem::exists(tokenizer_path + "/tokenizer.model")) {
+      // if 'tokenizer.model exists in the tokenizer_path, then it is a sentencepiece model
+      type = "SPM";
```

Comment:
This doesn't look right. SPM is not supported anymore.

Reply:
It's just a type check and only output an error message if the model is unsupported.